whois-api/src/main/java/net/ripe/db/whois/api/elasticsearch/ElasticIndexService.java

```diff
@@ -86,34 +99,54 @@ public boolean isEnabled() {
         return true;
     }
 
-    public void addEntry(RpslObject rpslObject) throws IOException {
-        final IndexRequest request = new IndexRequest(whoisIndex);
+    protected void addEntry(final RpslObject rpslObject) throws IOException {
+        if (!isElasticRunning()) {
+            return;
+        }
+
+        final IndexRequest request = new IndexRequest(whoisAliasIndex);
         request.id(String.valueOf(rpslObject.getObjectId()));
         request.source(json(rpslObject));
         client.index(request, RequestOptions.DEFAULT);
     }
 
-    public void deleteEntry(int objectId) throws IOException {
-        final DeleteRequest request = new DeleteRequest(whoisIndex, String.valueOf(objectId));
+    protected void deleteEntry(final int objectId) throws IOException {
+        if (!isElasticRunning()) {
+           return;
+        }
+
+        final DeleteRequest request = new DeleteRequest(whoisAliasIndex, String.valueOf(objectId));
         client.delete(request, RequestOptions.DEFAULT);
     }
 
-    public void deleteAll() throws IOException {
-        final DeleteByQueryRequest request = new DeleteByQueryRequest(whoisIndex);
+    protected void deleteAll() throws IOException {
+        if (!isElasticRunning()) {
+            return;
+        }
+
+        final DeleteByQueryRequest request = new DeleteByQueryRequest(whoisAliasIndex);
         request.setQuery(QueryBuilders.matchAllQuery());
 
         client.deleteByQuery(request, RequestOptions.DEFAULT);
     }
 
-    public long getWhoisDocCount() throws IOException {
-        final CountRequest countRequest = new CountRequest(whoisIndex);
+    protected long getWhoisDocCount() throws IOException {
+        if (!isElasticRunning()) {
+            throw new IOException("ES is not running");
```

Comment:
I suggest using `IllegalStateException` as a more generic (and unchecked) exception, this situation is not an IOException. However, IOException maybe fine too, if the caller is already handling it (rather than adding a new catch).

Reply:
👍